src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,162 @@
+
+export const PRESTART_STATUSES = {
+  notAttempted: 'not-attempted',
   active: 'active',
-  claim_received: 'claim_received', // eslint-disable-line camelcase
-  duplicate: 'duplicate',
-  expired: 'expired',
   pending: 'pending',
-  incomplete: 'incomplete'
-});
-
-
-// TODO: replace mock once ITF endpoint setup
-export function submitIntentToFile(formConfig, onChange) {
-  const { intentToFileUrl, formId, migrations, prefillTransformer } = formConfig;
-  let ITFStatus = ITFStatuses.pending;
-  return dispatch => {
-
-    onChange(ITFStatus);
-
-    const delay = (ms) => new Promise((resolve, reject) => {
-      ITFStatus = ITFStatuses.active;
-      onChange({ ITFStatus });
-
-      // TODO: if the backend handles resubmission, this check can be removed
-      if (ITFStatus === ITFStatuses.active) {
-        setTimeout(resolve, ms);
-      } else {
-        const errorMessage = 'Network request failed';
-        onChange({
-          ITFStatus,
-          errorMessage
-        });
-        reject(errorMessage);
-      }
-    });
-
-    return delay(2000);
+  none: 'none',
+  expired: 'expired',
+  created: 'created',
+  retrieved: 'retrieved',
+  renewed: 'renewed',
+  notRetrievedSaved: 'not-retrieved-saved',
+  notRetrievedNew: 'not-retrieved-new',
+  notCreated: 'not-created',
+  notRenewed: 'not-renewed',
+};
+
+export const prestartSuccessStatuses = new Set([PRESTART_STATUSES.created, PRESTART_STATUSES.retrieved, PRESTART_STATUSES.renewed]);
+
+export const prestartFailureStatuses = new Set([PRESTART_STATUSES.notCreated, PRESTART_STATUSES.notRenewed, PRESTART_STATUSES.notRetrievedNew, PRESTART_STATUSES.notRetrievedSaved]);
+
+export const prestartPendingStatuses = new Set([PRESTART_STATUSES.none, PRESTART_STATUSES.expired, PRESTART_STATUSES.pending]);
+
+export function setPrestartStatus(status) {
+  return {
+    type: PRESTART_STATUS_SET,
+    status
+  };
+}
+
+export function setPrestartData(data) {
+  return {
+    type: PRESTART_DATA_SET,
+    data
+  };
+}
+
+export function resetPrestartStatus() {
+  return {
+    type: PRESTART_RESET
+  };
+}
+
+export function resetPrestartDisplay() {
+  return {
+    type: PRESTART_DISPLAY_RESET
   };
 }
 
-//   apiRequest(
-//     `${intentToFileUrl}`,
-//     null,
-//     ({ data }) => {
-//       const ITFStatus = data.attributes.ITFStatus
-//       onChange({ ITFStatus });
-//       return ITFStatus === 'active';
-//     },
-//     ({ errors }) => {
-//       const errorMessage = 'Network request failed';
-//       onChange({
-//         ITFStatus,
-//         errorMessage
-//       });
-//       Raven.captureMessage(`vets_itf_error: ${errorMessage}`);
-//     }
-//   );
-// };
-// }
-/* eslint-enable */
+export const handleCheckSuccess = (data, dispatch) => {
+  let status;
+  let expirationDate;
+  const itfList = data.attributes.intentToFile;
+  // If the user does not have any existing ITFs, set none status
+  if (!itfList || (Array.isArray(itfList) && itfList.length === 0)) {
```

Comment:
I've been thinking about this and I'm wondering if the following might be a cleaner approach:

```js
if (!itfList.length) {
  return ITF_STATUSES.none;
}

// Grab the latest ITF
// This assumes some things about how ITFs are handled by EVSS (or VA writ large?)
const latestITF = getLatestITF(itfList);

// We only show information for active or expired ITFs
if (['active', 'expired'].includes(latestITF.status)) {
  return ITF_STATUSES.none;
}

const whichDate = latestITF.status === 'active' ? 'currentExpirationDate' : 'previousExpirationDate';
dispatch(setPrestartData({ [whichDate]: latestITF.expirationDate }));
return latestITF.status; // should be 'active' or 'expired', which should match properties in ITF_STATUSES
```

I'm not sure what about the approach you've got already that makes me uncomfortable, but maybe it's the level of `if`s. Does the above solve all the problems?

Reply:
Actually, now that I think about it, if we keep what you've got there, what happens if the latest ITF is `claim_received`, but there's an `expired` earlier in the history? I think `handleCheckSuccess` will return `expired` erroneously (since they successfully submitted a claim after that).

Reply:
Thanks @cvalarida, that's pretty clean, will update according to your recommendation.